ocs_ci/ocs/ui/base_ui.py

```diff
@@ -78,44 +78,67 @@ def __init__(self, driver: WebDriver):
             Path(self.dom_folder).mkdir(parents=True, exist_ok=True)
         logger.info(f"screenshots pictures:{self.dom_folder}")
 
-    def do_click(self, locator, timeout=30, enable_screenshot=False, copy_dom=False):
+    def do_click(
+        self,
+        locator,
+        timeout=30,
+        enable_screenshot=False,
+        copy_dom=False,
+        avoid_stale=False,
+    ):
         """
         Click on Button/link on OpenShift Console
 
         locator (set): (GUI element needs to operate on (str), type (By))
         timeout (int): Looks for a web element repeatedly until timeout (sec) happens.
         enable_screenshot (bool): take screenshot
         copy_dom (bool): copy page source of the webpage
+        avoid_stale (bool): if got StaleElementReferenceException, caused by reference to stale, cached element,
+        refresh the page once and try click again
+        * don't use when refreshed page expected to be different from initial page, or loose input values
```

Comment:
I did not get the meaning of this line from doc text, where are we checking that the refreshed page is same or different from the initial page? Isn't by default it's expected to be the same page?

Reply:
there are cases when you proceed with filling forms and refresh will clear your inputs. Or PopUp appears, it may disappear after refresh, or when we interacting with Topology and selecting the tab of chosen node, for instance Observe, it will be shown a default Resources after refresh. So shortly it depends on UX logic and must be validated by tester when he writes the test